jobs/pywikibot/scripts/updatestatistics.py

```diff
@@ -126,7 +125,7 @@ def get_trending_pages(service, profile_id):
 def update_list_of_pages(template, pages):
     template_page = pywikibot.Page(pywikibot.Link(template), pywikibot.Site())
     text = " <noinclude>This page is automatically generated. Changes will be overwritten, so '''do not modify'''.</noinclude>\n"
-    for p in pages:
+    for p in pages[:REQD_TEMPLATE_LEN]:
```

Comment:
I was a bit surprised by this truncation behaviour in the `update_...` function. I think it would be slightly clearer to do the truncation outside this function (ie. in the `main` method), so the only concern in here is actually updating the page.

Reply:
Ah, yes. I will move this into the main function. I didn't think that doing this in the get_popular_pages function was the right choice because (as in this case) we might need to have a larger set from the DB which we will prune based on our policy. So, I am getting all 50 results from the DB using that function.